lib/inc/internal/facts/windows/virtualization_resolver.hpp

```diff
@@ -29,6 +29,13 @@ namespace facter { namespace facts { namespace windows {
          * @return Returns the name of the hypervisor or empty string if no hypervisor.
          */
         virtual std::string get_hypervisor(collection& facts) override;
+        /**
+         * Gets the name of the cloud provider.
+         * @param facts The fact collection that is resolving facts.
+         * @return Returns the name of the cloud provider or empty string if no hypervisor.
+         */
+        virtual std::string get_cloud_provider(collection& facts) override;
```

Comment:
If you are going to declare that this subclass overrides the virtual function, you have to write an actual implementation. This is why appveyor is failing right now. Same goes for the Solaris one above. It might make more sense for now to leave out this declaration and let the default implementation be used on these non-linux platforms.